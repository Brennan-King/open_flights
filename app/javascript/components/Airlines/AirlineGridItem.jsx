import React from 'react';
import PropTypes from 'prop-types';
// eslint-disable-next-line no-unused-vars
import {BrowserRouter as Router, Link} from 'react-router-dom';
import './AirlineGridItem.scss';

const AIRLINE_GRID_ITEM_BUTTON_TEXT = 'View Airline';

const propTypes = {
  /** The url of the airline image. */
  imageUrl: PropTypes.string,
  /** The name of the airline. */
  name: PropTypes.string,
  /** The average score rating for the airline. */
  avgScore: PropTypes.number,
  /** The name formatted with dashes between words. */
  slug: PropTypes.string,
};

const propDefaults = {
  imageUrl: '',
  name: '',
  avgScore: 0,
  slug: '',
};

/**
 * Constructs an airline grid item to be placed within a grid.
 * @param {object} props the props used to configure the airline grid item
 * @return {node} a div containing the configured AirglinGridItem
 */
const AirlineGridItem = (props) => {
  const {imageUrl, name, avgScore, slug} = props;

  const airlineAverageScoreText = `Average Rating: ${avgScore}`;

  return (
    <div className="airline-grid-item-card">
      <div className="airline-grid-item-logo-container">
        <img className='airline-grid-item-logo' src={imageUrl} alt={name} />
      </div>
      <div className="airline-grid-item-name">{name}</div>
      <div className="airline-score">{airlineAverageScoreText}</div>
      <div className="airline-grid-item-link-container">
        <Link className="airline-grid-item-link" to={`/airlines/${slug}`}>
          {AIRLINE_GRID_ITEM_BUTTON_TEXT}
        </Link>
      </div>
    </div>
  );
};

AirlineGridItem.propTypes = propTypes;
AirlineGridItem.defaultProps = propDefaults;
export default AirlineGridItem;
