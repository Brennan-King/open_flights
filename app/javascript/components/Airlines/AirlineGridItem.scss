.airline-grid-item-card {
  border: 2px solid #000;
  border-radius: 5px;
  background: #3796cb;
  text-align: center;
  box-shadow: 10px 10px 5px grey;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  color: #fff;
}

.airline-grid-item-logo-container {
  width: 50px;
  text-align: center;
  margin-left: auto;
  margin-right: auto;
  padding-top: 10px;
}

.airline-grid-item-logo {
  height: 50px;
  width: 50px;
  border-radius: 100%;
  border: 1px solid #fff
}

.airline-grid-item-name {
  padding: 20px 0 10px 0;
}

.airline-grid-item-link-container {
  margin: 30px 0 20px 0;
  height: 50px;
}

.airline-grid-item-link {
  color: #fff;
  background: #000;
  border-radius: 4px;
  padding: 10px 50px;
  border: 1px solid #000;
  width: 100%;
  text-decoration: none;
}