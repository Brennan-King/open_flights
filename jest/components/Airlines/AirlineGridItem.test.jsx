/* eslint-disable max-len */
import 'jsdom-global/register';
import React from 'react';
import AirlineGridItem from '../../../app/javascript/components/Airlines/AirlineGridItem';
import {BrowserRouter, Router} from 'react-router-dom';
import {configure, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({adapter: new Adapter()});

/**
 * Test suite for AirlineGridItem component
 */
describe('AirlineGridItem', () => {
  it('should render an AirlineGridItem component with default props', () => {
    const component = mount(
        <Router history={new BrowserRouter().history}>
          <AirlineGridItem />
        </Router>,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render an AirlineGridItem component', () => {
    const component = mount(
        <Router history={new BrowserRouter().history}>
          <AirlineGridItem
            imageUrl="testImageUrl"
            name="testAirlineName"
            avgScore={5}
            slug="test-airline-name"
          />
        </Router>,
    );

    expect(component).toMatchSnapshot();
  });
});
