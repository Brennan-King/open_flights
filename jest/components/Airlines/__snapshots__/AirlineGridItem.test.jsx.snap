// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirlineGridItem should render an AirlineGridItem component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <AirlineGridItem
    avgScore={5}
    imageUrl="testImageUrl"
    name="testAirlineName"
    slug="test-airline-name"
  >
    <div
      className="airline-grid-item-card"
    >
      <div
        className="airline-grid-item-logo-container"
      >
        <img
          alt="testAirlineName"
          className="airline-grid-item-logo"
          src="testImageUrl"
        />
      </div>
      <div
        className="airline-grid-item-name"
      >
        testAirlineName
      </div>
      <div
        className="airline-score"
      >
        Average Rating: 5
      </div>
      <div
        className="airline-grid-item-link-container"
      >
        <Link
          className="airline-grid-item-link"
          to="/airlines/test-airline-name"
        >
          <LinkAnchor
            className="airline-grid-item-link"
            href="/airlines/test-airline-name"
            navigate={[Function]}
          >
            <a
              className="airline-grid-item-link"
              href="/airlines/test-airline-name"
              onClick={[Function]}
            >
              View Airline
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </div>
  </AirlineGridItem>
</Router>
`;

exports[`AirlineGridItem should render an AirlineGridItem component with default props 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <AirlineGridItem
    avgScore={0}
    imageUrl=""
    name=""
    slug=""
  >
    <div
      className="airline-grid-item-card"
    >
      <div
        className="airline-grid-item-logo-container"
      >
        <img
          alt=""
          className="airline-grid-item-logo"
          src=""
        />
      </div>
      <div
        className="airline-grid-item-name"
      />
      <div
        className="airline-score"
      >
        Average Rating: 0
      </div>
      <div
        className="airline-grid-item-link-container"
      >
        <Link
          className="airline-grid-item-link"
          to="/airlines/"
        >
          <LinkAnchor
            className="airline-grid-item-link"
            href="/airlines/"
            navigate={[Function]}
          >
            <a
              className="airline-grid-item-link"
              href="/airlines/"
              onClick={[Function]}
            >
              View Airline
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </div>
  </AirlineGridItem>
</Router>
`;
